bpf/bpf_sock.c

```diff
@@ -464,6 +477,44 @@ int sock4_post_bind(struct bpf_sock *ctx)
 }
 #endif /* ENABLE_NODEPORT || ENABLE_EXTERNAL_IP */
 
+#ifdef ENABLE_HEALTH_CHECK
+static __always_inline void sock4_auto_bind(struct bpf_sock_addr *ctx)
+{
+	ctx->user_ip4 = 0;
+	ctx_set_port(ctx, 0);
+}
+
+static __always_inline int __sock4_pre_bind(struct bpf_sock_addr *ctx,
+					    struct bpf_sock_addr *ctx_full)
+{
+	__u64 key = sock_local_cookie(ctx_full);
+	struct lb4_health val = {
+		.peer = {
+			.address	= ctx->user_ip4,
+			.port		= ctx_dst_port(ctx),
+			.proto		= ctx->protocol,
+		},
+	};
+	int ret;
+
+	ret = map_update_elem(&LB4_HEALTH_MAP, &key, &val, 0);
+	if (!ret)
+		sock4_auto_bind(ctx);
```

Comment:
Should we throw a bind error if ret != 0 here? Otherwise things are going to go a bit sideways later. If the pre_bind fails then the cilium health check tool can give some meaningful error.

Reply:
good idea, and it indeed will get propagated, done